rename_imports.py

```diff
@@ -0,0 +1,69 @@
+from LSP.plugin.core.file_watcher import FileWatcher
+from LSP.plugin.core.file_watcher import FileWatcherEvent
+from LSP.plugin.core.file_watcher import FileWatcherProtocol
+from LSP.plugin.core.typing import List, Type
+import os
+import sublime
+
+
+class HandleRenameImport(FileWatcherProtocol):
+    def __init__(self, window: sublime.Window, root_path: str, file_watcher: Type[FileWatcher]):
+        self.window = window
+        self.events = []  # type: List[FileWatcherEvent]
+        self.file_watcher = file_watcher.create(
+            root_path=root_path,
+            patterns=["**/*.ts", "**/*.tsx", "**/*.js", "**/*.jsx"],
+            events=["create", "delete"],
+            ignores=['**/node_modules/**'],
+            handler=self
+        )
+
+    def clean_up(self):
+        self.file_watcher.destroy()
+
+    def on_file_event_async(self, events: List[FileWatcherEvent]) -> None:
+        self.events.extend(events)
+        sublime.set_timeout_async(self._check_if_there_was_a_rename, 10)
+
+    def _check_if_there_was_a_rename(self):
+        if len(self.events) == 2:
+            create_event = next((event for event in self.events if event[0] == 'create'), None)
+            delete_event = next((event for event in self.events if event[0] == 'delete'), None)
+            if (create_event and delete_event):
+                # Rename detected
+                new_name = create_event[1]
+                old_name = delete_event[1]
+                self.prompt_rename(old_name, new_name)
+        self.events = []
+
+    def prompt_rename(self, old_name: str, new_name: str) -> None:
+        settings = sublime.load_settings("LSP-typescript.sublime-settings")
+        update_imports_on_file_move_setting = settings.get('settings', {}).get('updateImportsOnFileMove', "prompt")
+        if update_imports_on_file_move_setting == "always":
```

Comment:
Given the the logic for checking if the file was renamed can trigger false positives pretty easily, I think that there should be no "always" option as it seems dangerous. It should be a boolean option where true triggers the prompt.

And we should communicate both the old and new name ideally.

Reply:
Agree, 
I will remove the setting in favor of a boolean option.
I will use the existing ts setting for this `allowRenameOfImportPath?: boolean;`.
https://github.com/armanio123/TypeScript/blob/fbf840114ef38aa044c6b4840d5a43747bcf8a4e/src/server/protocol.ts#L3392